File: movie_rating_system/MovieRatingSystemTest.java
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class MovieRatingSystemTest {

    @BeforeEach
    public void setUp() {
     
        MovieRatingSystem.movies.clear();
        MovieRatingSystem.ratings.clear();
    }

    @Test
    public void testAddMovie() {
        String result = MovieRatingSystem.addMovie("Koto Aye");
        assertEquals("Movie 'Koto Aye' added!", result);
        assertTrue(MovieRatingSystem.movies.contains("Koto Aye"));
    }

    @Test
    public void testRateMovie() {
        MovieRatingSystem.addMovie("Koto Aye");
        String result = MovieRatingSystem.rateMovie("Koto Aye", 5.0);
        assertEquals("Rating added for 'Koto Aye' 5.0", result);
    }

    @Test
    public void testInvalidRating() {
        MovieRatingSystem.addMovie("Koto Aye");
        String result = MovieRatingSystem.rateMovie("Koto Aye", 10.0);
        assertEquals("Invalid rating! Enter a number from 1 to 5.", result);
    }

    @Test
    public void testViewAverageRatings() {
        MovieRatingSystem.addMovie("Inception");
        MovieRatingSystem.rateMovie("Inception", 5.0);
        MovieRatingSystem.rateMovie("Inception", 4.0);

        String expected = "Average Ratings:\nInception: 4.5";
        String actual = MovieRatingSystem.viewAverageRatings();

        assertEquals(expected, actual);
    }
}

File: movie_rating_system/movie_rating_system_test.py
from unittest import TestCase
from movie_rating_system import movies, ratings, add_movie, rate_movie, view_average_ratings

class TestMovieRatingSystem(TestCase):

    def setUp(self):
       
        movies.clear()
        ratings.clear()

    def test_add_movie(self):
        result = add_movie("Koto Aye")
        self.assertEqual(result, "Movie 'Koto Aye' added!")
        self.assertIn("Koto Aye", movies)

    def test_rate_movie(self):
        add_movie("Koto Aye")
        result = rate_movie("Koto Aye", 5)
        self.assertEqual(result, "Rating added for 'Koto Aye' 5")

    def test_invalid_rating(self):
        add_movie("Koto Aye")
        result = rate_movie("Koto Aye", 10)
        self.assertEqual(result, "Invalid rating! Enter a number from 1 to 5.")

    def test_view_average_ratings(self):
        add_movie("Inception")
        rate_movie("Inception", 5)
        rate_movie("Inception", 4)
        
        result = view_average_ratings()
        self.assertEqual(result, "Average Ratings:\nInception: 4.5")
